### PR TITLE
feat: ✨ Add Body Scroll Lock for Modals

### PR DESCRIPTION
**Pull request by GoOz**
Added the dependency [Body Scroll Lock](https://github.com/willmcpo/body-scroll-lock) to prevent background scrolling when Modal is displayed and scrollable. 
Essential especially on iOS but useful on other mobile OS and desktop.

For now, it's only used with Modal but could be extended to other components if needed.

### CODE REVIEW

File: react/Modal/index.jsx [hunk 1]
```diff
@@ -56,12 +58,12 @@ class ModalContent extends Component {
     this.scrollingContent.addEventListener('scroll', this.handleScroll, {
       passive: true
     })
-    document.body.classList.add('has-modal')
+    disableBodyScroll(document.querySelector(this.contentID))
```
Comment (Crash--) [inline react/Modal/index.jsx]: You don't need to add a new ID. You can use the already defined ref.

Something like : 
```
disableBodyScroll(this.scrollingContent)
```

To quote React documentation: 
```
 Refs are guaranteed to be up-to-date before componentDidMount or componentDidUpdate fires
```

**Comment by y-lohse** [comment]
I think removing the `has-modal` was an accident? It had been introduced [here](https://github.com/cozy/cozy-ui/pull/588) and well, alerts in modals on mobile are now broken :(

**Comment by y-lohse** [comment]
Ok :D Should I make a PR that re-adds the classname, or do we want to do this a different way?

**Comment by GoOz** [comment]
@y-lohse My mistake, I thought this classname had another purpose, something old that I already removed in the past. This should get back as it was before.
I was gonna fix/revert this this morning but if you're on it then go for it. :)

**Pull request merged**